File: backend/app/Models/Perfume.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Perfume extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'nama_parfum',
        'merek',
        'aroma',
        'ukuran',
        'konsentrasi',
        'status',
        'catatan',
        'image_url',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: frontend/lib/models/perfume.dart
class Perfume {
  int? id;
  String namaParfum;
  String merek;
  String aroma;
  String ukuran;
  String konsentrasi;
  String status;
  String catatan;
  String imageUrl;

  Perfume({
    this.id,
    required this.namaParfum,
    required this.merek,
    required this.aroma,
    required this.ukuran,
    required this.konsentrasi,
    required this.status,
    required this.catatan,
    required this.imageUrl,
  });

  factory Perfume.fromJson(Map<String, dynamic> json) {
    return Perfume(
      id: json['id'] as int?,
      namaParfum: json['nama_parfum']?.toString() ?? '',
      merek: json['merek']?.toString() ?? '',
      aroma: json['aroma']?.toString() ?? '',
      ukuran: json['ukuran']?.toString() ?? '',
      konsentrasi: json['konsentrasi']?.toString() ?? '',
      status: json['status']?.toString() ?? '',
      catatan: json['catatan']?.toString() ?? '',
      imageUrl: json['image_url']?.toString() ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nama_parfum': namaParfum,
      'merek': merek,
      'aroma': aroma,
      'ukuran': ukuran,
      'konsentrasi': konsentrasi,
      'status': status,
      'catatan': catatan,
      'image_url': imageUrl,
    };
  }
}
